src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3310,6 +3310,11 @@ public bool PackageExistsInPackagesFolder(PackageIdentity packageIdentity)
                 framework,
                 packages);
 
+            if (version != null)
+            {
+                packages = packages.Where(p => PrunePackageTree.MeetsVersionConstraints(p.Version, version, resolutionContext.VersionConstraints)).ToList();
```

Comment:
can you add a new api in `PrunePackageTree` class itself to apply VersionConstraints on a list of packages? something like: 
`public static IEnumerable<SourcePackageDependencyInfo> PruneByUpdateConstraints(IEnumerable<SourcePackageDependencyInfo> packages, VersionConstraints versionConstraints)`

or you can also use existing `PruneByUpdateConstraints` api... either way will be much cleaner and reusable.

Reply:
I added it to `PrunePackageTree` in the latest push